### ISSUE TITLE
Base suite packages are scanned every time if the base suite isn't itself in the config

### ISSUE THREAD

**Issue by iainlane** [ximion/appstream-generator#33]
After I fixed https://github.com/ximion/appstream-generator/pull/31, I noticed that every run scans all packages in `xenial`:

```
2016-11-15 10:30:26 - INFO: Scanning new packages for xenial-proposed/main [amd64]
2016-11-15 10:30:35 - INFO: Scanned xchat-gnome/1:0.30.0~git20141005.816798-0ubuntu9/amd64, could be interesting.
2016-11-15 10:30:35 - INFO: Scanned gnome-session-common/3.18.1.2-1ubuntu1/amd64, could be interesting.
2016-11-15 10:30:35 - INFO: Scanned transmission-gtk/2.84-3ubuntu3/amd64, could be interesting.
2016-11-15 10:30:35 - INFO: Scanned gnome-system-monitor/3.18.2-1/amd64, could be interesting.
```

and so on for all packages in `xenial`.

I think it's because [here](https://github.com/ximion/appstream-generator/blob/master/src/asgen/engine.d#L193) we would normally skip the package if it's been seen before, but we don't because it's not in `dstore`. AFAICS that is because [only the packages for the actual suite are processed](https://github.com/ximion/appstream-generator/blob/master/src/asgen/engine.d#L500).

Is that the bug? Do we need to process base suite packages? What do we need base suite packages for - just icons? If so, do they need to be in `seedContentsData` at all?

**Comment by iainlane** [comment]
Any thoughts? Just poked the logs and was reminded that this eats up a bit of time.

It'd make less work for the generator to drop the base suite from `seedContentsData`, but it would be safer to process base suite packages.

**Comment by iainlane** [comment]
I think we can do something like process the base suite in a separate pass in `seedContentsData`, but only for `cstore`, no `dstore` or `interestingFound`. Will look more after the holidays.

**Comment by iainlane** [comment]
Testing the referenced branch on `staging.appstream.ubuntu.com` now.

**Comment by ximion** [comment]
That patch looks good in itself, and it looks sane for fixing the issue - sorry that I haven't followed up on this, but I was (and still am) overly busy with other things. At least the Debian freeze is now, so that means a bit less stuff to do :) (if there are no RC bugs)

Btw, there will be a new asgen release in a relatively short time, thanks to https://bugzilla.gdcproject.org/show_bug.cgi?id=251 I ignored a failing CI so the last release has had some preventable bugs (patched in the 0.9.2 Debian package, but providing a sane upstream tarball for other distros would be useful). For the .3 release I also want to add the option to move Media/Html/etc. directories to custom locations (would make some rsync'ing unnecessary)

**Comment by iainlane** [comment]
Works well - saves quite a few minutes - thanks!
